pkgs/by-name/si/simg2img/package.nix

```diff
@@ -13,13 +19,42 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ zlib ];
 
-  makeFlags = [ "PREFIX=$(out)" "DEP_CXX:=$(CXX)" ];
+  makeFlags = [
+    "PREFIX=$(out)"
+    "DEP_CXX:=$(CXX)"
+  ];
+
+  patches = [
+    (fetchpatch {
+      name = "PR-38-Fix-for-Apple-Silicon.patch";
+      url = "https://github.com/anestisb/android-simg2img/commit/931df9dd83e7feea11197402c5b4e7ad489f4abf.patch";
+      hash = "sha256-+vHdx+nwaeaLf7a/hZSAVECc/Q5nQ1G2i0eTOVHb8Ks=";
+    })
+    (fetchpatch {
+      name = "PR-41-fix-calloc-argument-order-for-gcc-14-1.patch";
+      url = "https://github.com/anestisb/android-simg2img/commit/800987799591acd76d7d33853f665873aa5d63be.patch";
+      hash = "sha256-5dEmZY6/MiKbxQu1amR9F8pAFhX4T+10RNMTxKaVbc0=";
+    })
+    (fetchpatch {
+      name = "PR-41-fix-calloc-argument-order-for-gcc-14-2.patch";
+      url = "https://github.com/anestisb/android-simg2img/commit/bfcfd2cc3e249c0a1e46d466c57b1dc5ee352291.patch";
+      hash = "sha256-Ot5Uij/Je1gS0e9f5zP5+GPAIRl94vew87VMjd0J3ic=";
+    })
+    (fetchpatch {
+      name = "PR-41-fix-calloc-argument-order-for-gcc-14-3.patch";
+      url = "https://github.com/anestisb/android-simg2img/commit/4fcde3f0cebbc9466f1e73de04d82bc548dd1414.patch";
+      hash = "sha256-h9fr5H6hfi9SqKK9T7dvHiLK3wRKnsxkczwS71Ubh/g=";
+    })
```

Comment:
These are not merged yet. You can copy the postPatch from my PR instead

Reply:
When https://github.com/NixOS/nixpkgs/pull/372073 goes in, I'll drop it.

I feel great disturbance in the force, when for some review comments I get that I should use patches when they are available instead of `postPatch` commands. (sigh)

Reply:
That is indeed the correct action, if they are merged in the upstream repository.
Right now I can force push to my branch, and all the patches would disappear.

Reply:
Never mind, cherry-picked that patch instead.